### DISCUSSION TITLE
Setup help

### DISCUSSION THREAD

**Discussion by jakohans76** [Tyler-Keith-Thompson/CucumberSwift#21] [Q&A] [answered]
Hi,

As a user of Cucumber for many years with Ruby and .js, but a complete newbie with Xcode and Swift, I'm not able to get a setup working. Basically, I have no clue how to execute the feature and step definitions I have created. I got the impression that a func had to start with "test" for Xcode to be able to recognize it as a test, but I do not have any of those.

I cloned the CucumberSwiftExample repo and was also not able to get Xcode to recognize a test. It said 0 tests from the test tab.

I have a .feature file in the Feature folder (with reference) and I have a .swift file with the StepImplementation extension.

import Foundation
import XCTest
import CucumberSwift

extension Cucumber: StepImplementation {
    public var bundle: Bundle {
        class MoodagentUITests: XCTest {}
        return Bundle(for: MoodagentUITests.self)
    }
    public var continueAfterFailure:Bool {
        // continue testing although a step fails
        return true
    }
      
    public func setupSteps() {
        Given("^I am on the welcome page$") {matches, _ in
            print(matches[0])
            print("Moodagent 4 Life")
        }
    }
}

I would expect this to work but I'm also sure I'm missing something totally obvious.
I created a UITest target, added the CucumberSwift to the Podfile and did the pod install.

**Comment by Tyler-Keith-Thompson** [comment]
Hey thanks for reaching out! Sorry you've had some trouble getting things up and running. If it's possible for you to zip up a sample project or link to one on GitHub I'll be able to give direct help, but let me take a crack at a few things.

So the fact you've got 0 tests is telling, CucumberSwift on its own actually has 1 test it just includes to validate your gherkin. So this tells me it doesn't see your `.feature` files. Often times this happens because people have their features folder named something other than the default (which is case sensitive). Take a look at [this link](https://github.com/Tyler-Keith-Thompson/CucumberSwift/wiki/plist-options) which details an option you can add to your info.plist to give it the relative path. 

NOTE: By default it uses "Features" (case sensitive).

You do not have to add functions that start with the name "test". If you were just using the XCTest runner that's how it identifies tests, but CucumberSwift as a runner does it based off of those Given/When/Then/And/But global functions. You'll notice in your test explorer that the tests are purple (indicating they were added at runtime by CucumberSwift) vs the blue ones that XCTest finds that are prefixed with test.

Hopefully that helps! Let me know if that was in fact your problem, it's bitten enough people it's probably worth me adding something that does a case insensitive search for a features folder by default.

**Comment by jakohans76** [comment]
Hi again. Thanks for the fast feedback. And thanks for what seems to be an awesome Cucumber library. As a tester working with Cucumber for more than 11 years, I wanted to try a different approach to apps UI testing (no more Appium) and learn about Xcode and Swift. I almost quit the idea due to the other Cucumber libraries for Xcode. This looks really cool.

I was not able to get anything running by setting the FeaturesPath unfortunately. In regards to a .zip of my project, I believe we can just use the CucumberSwiftExample. I'm also getting 0 tests - no "purple" (or blue for that matter) tests are available. I've tried removing all pods and re-installing them. Same result. Can you think of any prerequisites I might miss? A dev account or something I'm not familiar with.

Have a great weekend.

**Reply by Tyler-Keith-Thompson** [reply]
Did you mean [this repo](https://github.com/c-st/CucumberSwiftExample)? That's one that somebody had thrown together to show me the same problem (in their case it was a features path issue). 

Tell you what, I'll create an official example repo and document each step this weekend. I'll share that here and ask you to clone it.

**Reply by jakohans76** [reply]
Much appreciated. Thanks a lot for taking the time to help an Apple newbie. I really want to use this library.

**Reply by Tyler-Keith-Thompson** [reply]
Alrighty, here's an official sample repo: https://github.com/Tyler-Keith-Thompson/CucumberSwiftSample/tree/main/Cocoapods

It's got detailed instructions for setup with CocoaPods and you're welcome to clone it and try it out! Hopefully that helps you (and others interested in this library) get started. Let me know if you've got any issues, especially if those steps still don't help you get it working.

**Reply by jakohans76** [reply]
Awesome. I'll try it tomorrow.

**Reply by jakohans76** [reply]
I couldn't wait so I went home and followed your sample guide. It works like a charm. Not sure what I have done wrong in my project but I'll try with your steps and get back to you. Once again thanks for your help and creating this awesome library and sample. And thanks for making my Sunday even greater.

**Reply by jakohans76** [reply]
I got it working with our own project. But I still have no clue what the difference was. Anyway, I really appreciate you taking the time to help. I will of course spread the word about this very cool Cucumber solution for Xcode.

**Comment by Tyler-Keith-Thompson** [comment, marked as answer]
Reposting cause I'm learning how to do GitHub discussions still and I don't like how it marked my answer, because it won't let me use a part of the thread as an answer:

Alrighty, here's an official sample repo: https://github.com/Tyler-Keith-Thompson/CucumberSwiftSample/tree/main/Cocoapods

It's got detailed instructions for setup with CocoaPods and you're welcome to clone it and try it out! Hopefully that helps you (and others interested in this library) get started. Let me know if you've got any issues, especially if those steps still don't help you get it working.
